### DISCUSSION TITLE
start_http_server with make_asgi_app?

### DISCUSSION THREAD

**Discussion by mecampbellsoup** [prometheus/client_python#1000] [Q&A] [answered]
Hi guys,

Quick question and hopefully not a dumb one but, nevertheless:

[`start_http_server`](https://github.com/prometheus/client_python/blob/2dcd17efd0ce2f0a1ad15cb3c150ffcdc42ced65/prometheus_client/exposition.py#L230) only works with [`make_wsgi_app`](https://github.com/prometheus/client_python/blob/2dcd17efd0ce2f0a1ad15cb3c150ffcdc42ced65/prometheus_client/exposition.py#L220) - is there anything preventing one from doing something similar with [`make_asgi_app`](https://github.com/prometheus/client_python/blob/2dcd17efd0ce2f0a1ad15cb3c150ffcdc42ced65/prometheus_client/asgi.py#L8) instead?

An alternative is to install [`prometheus-async`](https://github.com/hynek/prometheus-async/) which has its own [`start_http_server`](https://github.com/hynek/prometheus-async/blob/eda4a832608a46039819a1a6e02281f3b780843c/src/prometheus_async/aio/web.py#L88).

We have a FastAPI + uvicorn (ASGI) application and are looking to expose some Prometheus metrics for export.

**Comment by LordKa0S** [comment, marked as answer]
```python
from fastapi import FastAPI
import prometheus_client

app = FastAPI()
metrics_app = prometheus_client.make_asgi_app()
app.mount("/metrics", metrics_app)
```
_Originally posted by @pquentin in https://github.com/prometheus/client_python/issues/512#issuecomment-591866511_
